azurelinuxagent/common/protocol/wire.py

```diff
@@ -1297,7 +1303,7 @@ def get_artifacts_profile(self):
         artifacts_profile = None
 
         if self.has_artifacts_profile_blob():
-            blob = self.ext_conf.artifacts_profile_blob
+            blob = self.get_ext_conf().artifacts_profile_blob
```

Comment:
Why not repeat the same pattern like above - `ext_conf = self.get_ext_conf()`

Reply:
In the case above host_plugin was being used in multiple places. In this case this was the only occurrence of it so directly called it from here

Reply:
postpone?

Reply:
Done